src/mesh/Router.cpp

```diff
@@ -304,9 +307,31 @@ bool perhapsDecode(MeshPacket *p)
                 p->which_payloadVariant = MeshPacket_decoded_tag; // change type to decoded
                 p->channel = chIndex;                             // change to store the index instead of the hash
 
+                /*
+                if (p->decoded.portnum == PortNum_TEXT_MESSAGE_APP) {
+                    DEBUG_MSG("\n\n** TEXT_MESSAGE_APP\n");
+                } else if (p->decoded.portnum == PortNum_TEXT_MESSAGE_COMPRESSED_APP) {
+                    DEBUG_MSG("\n\n** PortNum_TEXT_MESSAGE_COMPRESSED_APP\n");
+                }
+                */
+
                 // Decompress if needed. jm
-                if (p->decoded.which_payloadVariant == Data_payload_compressed_tag) {
-                    // Decompress the file
+                if (p->decoded.portnum == PortNum_TEXT_MESSAGE_COMPRESSED_APP) {
+                    // Decompress the payload
+                    char compressed_in[Constants_DATA_PAYLOAD_LEN] = {};
+                    char decompressed_out[Constants_DATA_PAYLOAD_LEN] = {};
+                    int decompressed_len;
+
+                    memcpy(compressed_in, p->decoded.payload.bytes, p->decoded.payload.size);
+
+                    decompressed_len = unishox2_decompress_simple(compressed_in, p->decoded.payload.size, decompressed_out);
+
+                    // DEBUG_MSG("\n\n**\n\nDecompressed length - %d \n", decompressed_len);
+
+                    memcpy(p->decoded.payload.bytes, decompressed_out, decompressed_len);
+
```

Comment:
In case payload.bytes is used as a string for printing out using say printf, suggest adding \0 (ASCII 0) at position decompressed_len as unishox2_decompress_simple does not add \0.